src/test/java/com/ncc/aif/TestUtils.java

```diff
@@ -0,0 +1,495 @@
+        assertAndDump(testDescription, true);
+    }
+
+    /**
+     * This method will validate the model using the provided validator and will dump the model as TURTLE if
+     * either the validation result is unexpected or if the model is valid and forceDump is true. Thus, forceDump
+     * can be used to write all the valid examples to console.
+     *
+     * @param testName {@link String} containing the name of the test
+     * @param expected true if validation is expected to pass, false o/w
+     */
+    private void assertAndDump(String testName, boolean expected) {
+        final Resource report = validator.validateKBAndReturnReport(model);
+        final boolean valid = ValidateAIF.isValidReport(report);
+
+        // print model if result unexpected or if forcing (for examples)
+        // Swap comments following 2 lines if forceDump should ALWAYS dump output
+        // if (valid != expected || forceDump) {
+        if (valid != expected || (forceDump && expected)) {
+            System.out.println("\n----------------------------------------------\n" + testName + "\n\nAIF Model:");
+            RDFDataMgr.write(System.out, model, RDFFormat.TURTLE_PRETTY);
+        }
+
+        // fail if result is unexpected
+        if (valid != expected) {
+            // only print output if there is any
+            if (!valid) {
+                System.out.println("\nFailure:");
+                RDFDataMgr.write(System.out, report.getModel(), RDFFormat.TURTLE_PRETTY);
+            }
+            fail("Validation was expected to " + (expected ? "pass" : "fail") + " but did not");
+        }
+    }
+}
+
+/**
+ * An extension of the TestUtils that supports testing restricted AIF.  Usage and features are the same as TestUtils.
+ */
+class NistTestUtils extends TestUtils {
+
+    /**
+     * Constructor for utilities for testing restricted AIF functionality.
+     *
+     * @param annotationNamespace namespace to use with URIs
+     * @param validator           an AIF validator instantiated based on the caller's ontology and desired NIST restrictions
+     * @param forceDump           whether or not to force dumping of models prior to validation
+     */
+    NistTestUtils(String annotationNamespace, ValidateAIF validator, boolean forceDump) {
+        super(annotationNamespace, validator, forceDump);
+    }
+
+    /**
+     * Call before each test.  Returns a new, empty model with standard AIF namespaces.
+     *
+     * @return a new model with which to start a test; caller may wish to add prefixes for the ontology and annotation
+     */
+    Model startNewTest() {
+        Model model = super.startNewTest();
+        // NIST tests always need type assertion justifications
+        addSourceDocumentToJustification(super.getTypeAssertionJustification(), getDocumentName());
+        return model;
+    }
+
+    Resource makeValidJustification() {
+        return makeValidJustification("sourceDocument");
```

Comment:
Maybe it could, but I also kind of wanted the source documents to "stand out" in the generated AIF.  Also, I wasn't sure about the restrictions related to source documents, so I made a way to keep the source documents the same-- using this method, `makeValidJustification()`-- and one where you could specify a source doc in case you wanted them to be different, `makeValidJustification(String)`.

That said, if both of these reasons are weak or invalid, let me know and I can change it to use `getDocumentName()`.

Reply:
Oh, also notice that this method isn't actually used at the moment...I haven't gone back and hacked up the Nist tests to use this method.

Reply:
Haha, well, if this method isn't needed, should we get rid of it? ;)

Reply:
Well, it isn't *used*...whether or not it's *needed* (or at least *desirable*) is a separate question.  ;)

Reply:
Haha, fair enough. This kinda goes along with our discussion about justifications, but if you want to be able to create valid justifications with a known, static document string, then this looks good

Reply:
It now just calls `getDocumentName()`.